cpp/include/cudf/detail/utilities/cuda.cuh

```diff
@@ -74,9 +74,10 @@ class grid_1d {
    * @param num_threads_per_block The number of threads per block
    * @return thread_index_type The global thread index
    */
-  static constexpr thread_index_type global_thread_id(thread_index_type thread_id,
-                                                      thread_index_type block_id,
-                                                      thread_index_type num_threads_per_block)
+  CUDF_HOST_DEVICE static constexpr thread_index_type global_thread_id(
+    thread_index_type thread_id,
+    thread_index_type block_id,
+    thread_index_type num_threads_per_block)
```

Comment:
Is this ever called on the host? Not sure what that use case would be.

Reply:
Good question, probably not. I'll switch it and make sure things compile.